app/main.py

```diff
@@ -1,8 +1,25 @@
 class Person:
-    # write your code here
-    pass
+
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_list = []
+
+    for person_data in people:
+        person = Person(name=person_data["name"], age=person_data["age"])
+        person_list.append(person)
+
+    for person_data in people:
+        person = Person.people[person_data["name"]]
+        if "wife" in person_data and person_data["wife"] is not None:
+            person.wife = Person.people[person_data["wife"]]
+        elif "husband" in person_data and person_data["husband"] is not None:
+            person.husband = Person.people[person_data["husband"]]
```

Comment:
Consider using `and` logical operator to avoid nested `if` and `elif` statements. This will make the code cleaner and more readable. For example, you can combine the conditions for 'wife' and 'husband' into a single line using `or`.